src/populateExchange.ts

```diff
@@ -251,6 +256,14 @@ export const addFragmentsToQuery = (
         },
       },
       Document: {
+        enter: node => {
+          node.definitions.reduce((set, definition) => {
```

Comment:
Adds the existing fragment definitions to a Set for subsequent checks

Reply:
I'd go for a `forEach` as we're mutating the set and it's already in scope

Reply:
We’ll have to go for a `forEsch` here actually as it’s a `Set` and IE11 doesn’t implement the iterable spec :)

Reply:
It's not iterating on a Set – it's reducing into one

Reply:
oh true! mahbad